packages/google_maps_flutter/ios/Classes/GoogleMapsPlugin.m

```diff
@@ -1,16 +1,404 @@
+// Forward declarations.
+
+static id positionToJson(GMSCameraPosition* position);
+static double toDouble(id json);
+static CLLocationCoordinate2D toLocation(id json);
+static GMSCameraPosition* toOptionalCameraPosition(id json);
+static GMSCoordinateBounds* toOptionalBounds(id json);
+static GMSCameraUpdate* toCameraUpdate(id json);
+static void writeMapOptions(id json, id<FLTGoogleMapOptionsSink> sink);
+static void writeMarkerOptions(id json, id<FLTGoogleMapMarkerOptionsSink> sink,
+                               NSObject<FlutterPluginRegistrar>* registrar);
+
+// GoogleMaps plugin implementation
+
+@implementation FLTGoogleMapsPlugin {
+  NSObject<FlutterPluginRegistrar>* _registrar;
+  FlutterMethodChannel* _channel;
+  NSMutableDictionary* _mapControllers;
+}
 
-@implementation FLTGoogleMapsPlugin
 + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
   FlutterMethodChannel* channel =
       [FlutterMethodChannel methodChannelWithName:@"plugins.flutter.io/google_maps"
                                   binaryMessenger:[registrar messenger]];
-  FLTGoogleMapsPlugin* instance = [[FLTGoogleMapsPlugin alloc] init];
+  FLTGoogleMapsPlugin* instance =
+      [[FLTGoogleMapsPlugin alloc] initWithRegistrar:registrar channel:channel];
   [registrar addMethodCallDelegate:instance channel:channel];
 }
 
+- (instancetype)initWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar
+                          channel:(FlutterMethodChannel*)channel {
+  self = [super init];
+  if (self) {
+    _registrar = registrar;
+    _channel = channel;
+    _mapControllers = [NSMutableDictionary dictionaryWithCapacity:1];
+  }
+  return self;
+}
+
 - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
-  result(FlutterMethodNotImplemented);
+  if ([call.method isEqualToString:@"init"]) {
+    for (FLTGoogleMapController* controller in _mapControllers.allValues) {
+      [controller removeFromView];
+    }
+    [_mapControllers removeAllObjects];
+    result(nil);
+  } else if ([call.method isEqualToString:@"createMap"]) {
+    NSDictionary* options = call.arguments[@"options"];
+    GMSCameraPosition* camera = toOptionalCameraPosition(options[@"cameraPosition"]);
+    FLTGoogleMapController* controller =
+        [FLTGoogleMapController controllerWithWidth:toDouble(call.arguments[@"width"])
+                                             height:toDouble(call.arguments[@"height"])
+                                             camera:camera];
+    _mapControllers[controller.mapId] = controller;
+    writeMapOptions(options, controller);
+    UIView* flutterView = [UIApplication sharedApplication].delegate.window.rootViewController.view;
+    [controller addToView:flutterView];
+    controller.delegate = self;
+    result(controller.mapId);
+  } else if ([call.method isEqualToString:@"showMapOverlay"]) {
+    FlutterError* error;
+    FLTGoogleMapController* controller = [self mapFromCall:call error:&error];
```

Comment:
Perhaps the reporting of `error` to `result` could be extracted into mapFromCall?

Or you could collect all the calls with a map id inside one else-branch